sdk/ai/azure-ai-inference/azure/ai/inference/_configuration.py

```diff
@@ -55,6 +55,8 @@ def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, "TokenCr
     def _infer_policy(self, **kwargs):
         if isinstance(self.credential, AzureKeyCredential):
             return policies.AzureKeyCredentialPolicy(self.credential, "Authorization", prefix="Bearer", **kwargs)
+        if isinstance(self.credential, AzureKeyCredential):
+            return policies.AzureKeyCredentialPolicy(self.credential, "api-key", **kwargs)
```

Comment:
Is there a way to determine whether to pass the API key with Authorization vs api-key? I'm wondering if we should patch this code.

Reply:
Per our discussion, I pushed another commit with changes to _patch.py to make sure two headers are set simultaneously when you do key auth: "Authrotization: Bearer <key>" and "api-key: <key", such that we target both MaaS/MaaP and AOAI endpoints.

Reply:
Please have a look at _patch.py for the __init__ method of the clients. I added this code:

```python
        # For AzureKeyCredential, _client.py is going to add an authentication_policy that results in the
        # "Authorization: Bearer <api_key>" header, to support Serverless API (aka MaaS) endpoints and
        # Managed Compute (aka MaaP) endpoints. We are also adding "api-key: <api-key>" header, to support
        # Azure OpenAI service, so the application does not need to explicitly set this header.
        if isinstance(credential, AzureKeyCredential):
            headers = kwargs.pop("headers", {})
            if "api-key" not in headers:
                headers["api-key"] = credential.key
            kwargs["headers"] = headers
```

Note that I could not do this be just redefining `authentication_policy` in kwargs, since that one can only point to ONE policy. I needed to be able to point to two auth policies in order for the auto-gen code to create to different auth headers. So I had to explicitly do this via setting HTTP request header directly.